Allow nikic/php-parser ^5.0.0

gazben opened the pull request:
The `nikic/php-parser` package is now upgraded to 5.0. According to the changelog, this version drops the php 5 parsing and requires php 7.4.

There is automated MR regarding this topic: https://github.com/maglnet/ComposerRequireChecker/pull/484

But that MR takes a more radical approach by not allowing the v4 version.

Changelog: https://github.com/nikic/PHP-Parser/releases/tag/v5.0.0

Ocramius commented:
This patch can only be merged if CI also runs the newest dependencies: "allowing" them is insufficient, and #484 highlights the associated failures

gazben commented:
Okay, in this case I'll update the MR in the next release of `vimeo/psalm` is released: https://github.com/vimeo/psalm/blob/master/composer.json#L36

renovate[bot] committed:
Update dependency phpunit/phpunit to ^10.5.31

| datasource | package         | from    | to      |
| ---------- | --------------- | ------- | ------- |
| packagist  | phpunit/phpunit | 10.5.30 | 10.5.31 |

renovate[bot] committed:
Update dependency phpstan/phpstan to ^1.12.1

| datasource | package         | from   | to     |
| ---------- | --------------- | ------ | ------ |
| packagist  | phpstan/phpstan | 1.12.0 | 1.12.1 |

renovate[bot] committed:
Update dependency phpunit/phpunit to ^10.5.32

| datasource | package         | from    | to      |
| ---------- | --------------- | ------- | ------- |
| packagist  | phpunit/phpunit | 10.5.31 | 10.5.32 |

renovate[bot] committed:
Update dependency phpstan/phpstan to ^1.12.2

| datasource | package         | from   | to     |
| ---------- | --------------- | ------ | ------ |
| packagist  | phpstan/phpstan | 1.12.1 | 1.12.2 |

renovate[bot] committed:
Lock file maintenance

renovate[bot] committed:
Update dependency vimeo/psalm to ^5.26.0

| datasource | package     | from   | to     |
| ---------- | ----------- | ------ | ------ |
| packagist  | vimeo/psalm | 5.25.0 | 5.26.0 |

renovate[bot] committed:
Update all non-major dependencies

| datasource | package         | from    | to      |
| ---------- | --------------- | ------- | ------- |
| packagist  | phpstan/phpstan | 1.12.2  | 1.12.3  |
| packagist  | phpunit/phpunit | 10.5.32 | 10.5.33 |

renovate[bot] committed:
Update dependency vimeo/psalm to ^5.26.1

| datasource | package     | from   | to     |
| ---------- | ----------- | ------ | ------ |
| packagist  | vimeo/psalm | 5.26.0 | 5.26.1 |

renovate[bot] committed:
Lock file maintenance

renovate[bot] committed:
Lock file maintenance

renovate[bot] committed:
Update dependency phpunit/phpunit to ^10.5.34

| datasource | package         | from    | to      |
| ---------- | --------------- | ------- | ------- |
| packagist  | phpunit/phpunit | 10.5.33 | 10.5.34 |

Ocramius commented:
Diff no longer contains `"nikic/php-parser": "^5",`?

gazben commented:
I thought that with the new release of `vimeo/psalm` the dependency is upgraded and I rebased the branch, but I learned quickly that this is not the case. I'm waiting for this https://github.com/vimeo/psalm/pull/11035

Sorry for the confusion

fredden commented:
According to https://github.com/vimeo/psalm/issues/11112#issuecomment-2610564516, `vimeo/psalm` "v6 will be released sometime this month." which includes support for `nikic/php-parser:^5.0`.

vjik commented:
Psalm 6 released. Is there anything else beed for nikic/php-parser 5 support?

gazben commented:
The problem is the `roave/infection-static-analysis-plugin` dependency.

Ocramius commented:
FYI: https://github.com/Roave/infection-static-analysis-plugin/pull/499

gazben committed:
Use psalm 6.0

gazben commented:
Thanks, updated with the new release.

Ocramius commented:
Sounds like the parser did indeed break a lot here. This run shows a ton of broken property access issues:

https://github.com/maglnet/ComposerRequireChecker/actions/runs/12984968315/job/36209988057?pr=545

Ocramius commented:
Done in #563

Thanks anyweay, @gazben!

gazben commented:
Sorry I din't have time to port it. I hope I can try out the new version

Ocramius commented:
@gazben it's fine, it's merged + released :-)
